cmd/kubelet/app/options/options.go

```diff
@@ -519,6 +519,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.BoolVar(&c.ProtectKernelDefaults, "protect-kernel-defaults", c.ProtectKernelDefaults, "Default kubelet behaviour for kernel tuning. If set, kubelet errors if any of kernel tunables is different than kubelet defaults.")
 	fs.StringVar(&c.ReservedSystemCPUs, "reserved-cpus", c.ReservedSystemCPUs, "A comma-separated list of CPUs or CPU ranges that are reserved for system and kubernetes usage. This specific list will supersede cpu counts in --system-reserved and --kube-reserved.")
 	fs.StringVar(&c.TopologyManagerScope, "topology-manager-scope", c.TopologyManagerScope, "Scope to which topology hints applied. Topology Manager collects hints from Hint Providers and applies them to defined scope to ensure the pod admission. Possible values: 'container', 'pod'.")
+	fs.Var(cliflag.NewMapStringStringNoSplit(&c.TopologyManagerPolicyOptions), "topology-manager-policy-options", "A set of key=value Topology Manager policy options to use, to fine tune their behaviour. If not supplied, keep the default behaviour.")
```

Comment:
@derekwaynecarr I've sort of lost track of how kubelet is managing the transition to a config file, but would it make sense to just have new structured options like this go into the config file, rather than also adding immediately-deprecated flags for them?

Reply:
Pending @derekwaynecarr 's response, I see that the following PRs added kubelet flags like this in the past year:
https://github.com/kubernetes/kubernetes/commit/0064010cddfa009fe16ae23fcd0c57f4f15d227c
https://github.com/kubernetes/kubernetes/commit/5983850dbc3cd40382a8c892433ffdc796bd29dd

And I was personally involved in reviewing / approving a PR in April 2021 that added a similar flag to this one (for the CPUManager, rather than the TopologyManager as in this case):
https://github.com/kubernetes/kubernetes/commit/cc76a756e409d08959b8b6e623e317619399f015

So there is at least precedent for doing this recently.

Reply:
@liggitt I think its fine to have this flag as its the same practice we followed for cpu manager policy options and it was the approach that was litigated during the kep process for this feature so changing this now doesn't feel right to me and is not different from past precedent. 

see: https://github.com/kubernetes/enhancements/pull/3549/files#r980945432

if we get a future kep, i will take this advice into account so we can avoid more flag proliferation.